### PR TITLE
Coin Gecko new arrival detection

### LINKED ISSUE #16: Turn new arrivals detection in coingecko api data into background service
## Description
Currently new arrival detection from coingecko data only works on the FE while that page is opened in a browser. I want to get notifications about new arrivals at CG platform rather than having to have an application page opened continuously in my browser.

## Acceptance Criteria
1. Set up coin gecko new arrivals detection as a background service.
2. Hook the notification flow/service into it too.

### PR DESCRIPTION
**Pull request by Shapokhan**
fixes #16

### CODE REVIEW

File: server/coin-gecko-scraper.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        this.incomingCoins.map(coin => {
+            if (!this.storedCoinList[`${coin.id}`]) {
+                message = `New Gecko Coin found, Id =  ${coin.id}, and Name = ${coin.name}`;
+                this.notifyOnDiscord(message);
+                this.updateCoinList(coin);
+            }
+        })
+
+    }
+    notifyOnDiscord(msg) {
```
Comment (omeryousaf) [inline server/coin-gecko-scraper.js:51]: @Shapokhan this method is being duplicated in the two scraping services, shouldn't we factor it out into a separate file and reuse it at both places ?

Comment (Shapokhan) [reply server/coin-gecko-scraper.js]: @omeryousaf Bhai moved this function to commen-functions.js(newly created) file.

File: server/coin-gecko-scraper.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const rp = require('request-promise');
+const { notifyOnDiscord } = require('./commenFunctions');
```
Comment (omeryousaf) [inline server/coin-gecko-scraper.js:2]: @Shapokhan `common` k spellings...

File: server/app.js [hunk 3]
```diff
@@ -2,14 +2,18 @@ const express = require('express');
...
 const app = express();
 const port = 3000;
+const CoinGeckoScraper = require('./coin-gecko-scraper');
 const CoinScraper = require('./coin-scraper');
 const { CUTOFF_DATE } = require('./constants');
 
 app.use(express.static('dist'));
 
 const scraper = new CoinScraper();
+const coineGeckoScraper = new CoinGeckoScraper();
```
Comment (omeryousaf) [inline server/app.js:12]: @Shapokhan `coin` k spellings...

**Review by omeryousaf — CHANGES_REQUESTED** [review]
@Shapokhan the project is crashing with following error:
```
internal/modules/cjs/loader.js:968
  throw err;
  ^

Error: Cannot find module './commenFunctions'
Require stack:
- /Users/omeryousaf/Documents/Projects/tests/sasta-miner/server/coin-gecko-scraper.js
```

File: server/coin-gecko-scraper.js (hunk 2 above)
Comment (Shapokhan) [reply server/coin-gecko-scraper.js]: @omeryousaf bhai fixed.

File: server/app.js (hunk 3 above)
Comment (Shapokhan) [reply server/app.js]: fixed. my poor mistakes.

**Review by omeryousaf — CHANGES_REQUESTED** [review]

**Review by omeryousaf — APPROVED** [review]

**Review by omeryousaf — CHANGES_REQUESTED** [review]
@Shapokhan can u plz look into the two issues highlighted in this review ?

File: server/coin-gecko-scraper.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                if (dictionarySize === 0) {
+                    this.storeCoins()
+                }
+                else {
+                    console.log('--Searching for new coin--')
+                    this.checkNewCoin()
+                }
+            }
+            catch (error) {
+                console.log(error)
```
Comment (omeryousaf) [inline server/coin-gecko-scraper.js:32]: @Shapokhan we should log time with error too.

File: Readme.md [hunk 5]
```diff
@@ -15,3 +15,7 @@
 4. run `yarn build`
 5. run `pm2 start server/app.js`
 6. open `http://<remote_server_ip>:3000`
+
+## To run unit tests
+1. cd into the server/ folder
```
[the remark this answers is not in the document]
Comment (omeryousaf) [reply Readme.md:20]: @Shapokhan first of these ☝️ issues is fixed but second one is still pending. The unit test added to the pr also highlights that issue.

File: server/coin-gecko-scraper.js [hunk 6]
```diff
@@ -0,0 +1,60 @@
+const rp = require('request-promise');
```
Comment (omeryousaf) [inline server/coin-gecko-scraper.js:1]: @Shapokhan good to see that you achieved in just about 60 LOC (lines of code) pretty much the same functionality that [i did in about 90 LOC](https://github.com/omeryousaf/sasta-miner/blob/334d339d26dcb618a064299c83e9583e7f8105f2/server/coin-scraper.js). 👍

File: Readme.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Shapokhan) [reply Readme.md:20]: @omeryousaf couldn't find time for 2nd. I Will look on the 2nd day of eid inshallah.

[the remark this answers is not in the document]
Comment (Shapokhan) [reply Readme.md:20]: @omeryousaf according to my local(Desi Type) test, It is sending 2 notification when 2 coins found as a new coin. Anyhow, I need your guidance on "How to run unit test script".

File: server/coin-gecko-scraper.js (hunk 6 above)
Comment (Shapokhan) [reply server/coin-gecko-scraper.js:2]: I get more confidence when you appreciate me. Thank you.

File: Readme.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Shapokhan) [reply Readme.md:20]: @omeryousaf Bhai fixed the second issue as well. Kindly have a look.

[the remark this answers is not in the document]
Comment (omeryousaf) [reply Readme.md:20]: now it does seem to be fixed @Shapokhan! merging this pr now.

**Review by omeryousaf — APPROVED** [review]

**Pull request merged**